assets/js/sharedClientFunctions.js

```diff
@@ -80,11 +81,10 @@ module.exports = {
       });
     };
     $scope.deleteRef = function (id) {
-      var current_user = $scope.refUser || $scope.user;
       var url = "/reference/delete";
       io.socket.post(url, {refId: id}, function (data, res) {
         if (res.statusCode === 200) {
-          current_user.references = current_user.references.filter(function (ref) {
+          $scope[current_user].references = $scope[current_user].references.filter(function (ref) {
```

Comment:
That's sort of what we're doing now. It was just repetitive typing out `functionName($scope.refUser || $scope.user` all the time, so I wanted to abstract out `$scope.refUser || $scope.user` to a variable. However, this was complicated by the fact that `$scope.refUser` is defined asynchronously, so saying something like `var current_user = $scope.refUser || $scope.user` wouldn't work because neither is defined at that point.